action.yaml

```diff
@@ -72,15 +72,17 @@ runs:
           PRE_COMMIT_ARGS="--files ${{ steps.changed-files.outputs.all_changed_files }}"
           echo "PRE_COMMIT_ARGS=$PRE_COMMIT_ARGS" >> "$GITHUB_ENV"
         fi
-    - name: Pre-commit (action)
-      # Same as above, this will install and run pre-commit for us
+    - name: Install Pre-commit
       if: env.PRE_COMMIT_BIN == null && steps.pre-commit-config.outputs.exists != 'false'
-      uses: pre-commit/action@v3.0.1
-      with:
-        extra_args: ${{ env.PRE_COMMIT_ARGS }}
+      run: |
+        pip install pre-commit
+        pip freeze --local
+      shell: bash
+    - run: pre-commit run --show-diff-on-failure --color=always ${{ inputs.extra_args }}
+      shell: bash
```

Comment:
@armand-sauzay I tried this change here, since updating the python version was not working using the `pre-commit` action. Let me know if this looks good to you!

For now I had to manually reference my branch here to test things https://github.com/open-turo/actions-gha/compare/f/use_new_action_pre_commit?expand=1

I think ideally we should use the python version of the consumer repo if there is one (or allow to specify it). What do you think?

Reply:
yes that should work (and ideally as you say using the python version of the consumer repo would be even better). `pip freeze --local` is probably optional but fine to keep for debug purposes

It's very similar to what the `pre-commit/action` does (with the cache that they use to speed up workflows). 

I think using cache (also since unused caches get deleted if unused since 7+ days) could be another low hanging fruit if we can bare the cost of gha caches.